packages/bulkapi/src/controllers/BulkApi.ts

```diff
@@ -2,48 +2,85 @@
 // - unpack the request
 // - call controller to do actual work
 // - pack response
-
+import { getLanguageRegistry } from "@lionweb/repository-languages"
 import { Request, Response } from "express"
 import { LionWebJsonChunk, LionWebValidator } from "@lionweb/validation"
-import { LIONWEB_BULKAPI_WORKER } from "./LionWebBulkApiWorker.js"
-import {logger} from "../logging.js";
+import { BulkApiContext } from "../BulkApiContext.js"
+import { logger } from "@lionweb/repository-dbadmin"
 
-export interface LionWebBulkApi {
+export interface BulkApi {
     partitions: (req: Request, res: Response) => void
+    createPartitions: (req: Request, res: Response) => void
+    deletePartitions: (req: Request, res: Response) => void
     store: (req: Request, res: Response) => void
     retrieve: (req: Request, res: Response) => void
 }
 
-class LionWebBulkApiImpl implements LionWebBulkApi {
+export class BulkApiImpl implements BulkApi {
+    
+    constructor(private ctx: BulkApiContext) {
+    }
     /**
      * Bulk API: Get all partitions (nodes without parent) from the repo
      * @param req no `parameters` or `body`
      * @param res The list of all partition nodes, without children or annotations
      */
-    async partitions(req: Request, res: Response): Promise<void> {
+    partitions = async (req: Request, res: Response): Promise<void> => {
         logger.requestLog(` * partitions request received, with body of ${req.headers["content-length"]} bytes`)
-        const result = await LIONWEB_BULKAPI_WORKER.bulkPartitions()
-        res.send(result)
+        const result = await this.ctx.bulkApiWorker.bulkPartitions()
+        res.status(result.status)
+        res.send(result.queryResult)
+    }
+
+    createPartitions = async (req: Request, res: Response): Promise<void> => {
+        const chunk: LionWebJsonChunk = req.body
+        const validator = new LionWebValidator(chunk, getLanguageRegistry())
+        validator.validateSyntax()
+        validator.validateReferences()
+        if (validator.validationResult.hasErrors()) {
+            // console.log("STORE VALIDATION ERROR " + validator.validationResult.issues.map(issue => issue.errorMsg()))
```

Comment:
Can we remove this commented out code or alternatively use the logger?

Reply:
Will remove it